src/compile/selection/assemble.ts

```diff
@@ -138,6 +138,34 @@ export function assembleLayerSelectionMarks(model: LayerModel, marks: any[]): an
   return marks;
 }
 
+export function assembleLegendSelection(model: Model, part: string, value: any) {
+  // Prototype for Single Selection - Single Field Projected Legend
+  let hasLegend = false;
+  let field;
+  let store;
+  forEachSelection(model, selCmpt => {
+    if (selCmpt.legend) {
+      hasLegend = true;
+      // Only single field for now
+      // To do: Refactor for multiple fields projection
+      field = selCmpt.project[0].field;
+      const vname = varName(selCmpt.name);
+      store = stringValue(vname + STORE);
+    }
+  });
+
+  if (hasLegend) {
+    const newValue = value ? value : {opacity: {value: 0.9}};
+    // To do : Add test case for legends and symbols
+    newValue.opacity = [
+      {test: `!(length(data(${store}))) || vlSelectionTest(${store}, {${field}: datum.value})`, ...newValue.opacity},
```

Comment:
I recommend using the `assembleSelectionPredicate` method to generate this test string, so that we're not duplicating important logic. That method may require some minor modification to support the custom second argument to `vlSelectionTest`.